src/libraries/System.Security.Cryptography.Cose/ref/System.Security.Cryptography.Cose.cs

```diff
@@ -70,12 +70,16 @@ public sealed partial class CoseSign1Message : System.Security.Cryptography.Cose
         [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
         public static byte[] Sign(System.ReadOnlySpan<byte> content, System.Security.Cryptography.AsymmetricAlgorithm key, System.Security.Cryptography.HashAlgorithmName hashAlgorithm, System.Security.Cryptography.Cose.CoseHeaderMap? protectedHeaders = null, System.Security.Cryptography.Cose.CoseHeaderMap? unprotectedHeaders = null, bool isDetached = false) { throw null; }
         [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
+        public static System.Threading.Tasks.Task<byte[]> SignAsync(System.IO.Stream detachedContent, System.Security.Cryptography.AsymmetricAlgorithm key, System.Security.Cryptography.HashAlgorithmName hashAlgorithm, System.Security.Cryptography.Cose.CoseHeaderMap? protectedHeaders = null, System.Security.Cryptography.Cose.CoseHeaderMap? unprotectedHeaders = null) { throw null; }
```

Comment:
We should also add synchronous versions of the Stream-accepting methods.  As you saw in the tests, sometimes you are in a synchronous callback/override and need to do work.

```C#
public static byte[] Sign(System.IO.Stream detachedContent, System.Security.Cryptography.AsymmetricAlgorithm key, System.Security.Cryptography.HashAlgorithmName hashAlgorithm, System.Security.Cryptography.Cose.CoseHeaderMap? protectedHeaders = null, System.Security.Cryptography.Cose.CoseHeaderMap? unprotectedHeaders = null) { throw null; }

public bool Verify(System.Security.Cryptography.AsymmetricAlgorithm key, System.ReadOnlySpan<byte> detachedContent) { throw null; }
```

Reply:
The reason I omitted sync overloads is because I think most of the stream scenarios can be satisfied with SignAsync/VerifyAsync (I may be wrong), it feels natural that if you have a large content you should compute the message in a separate thread.

If we add `byte[] Sign(Stream)` should we also add `bool TrySign(Stream)`? And what about `CancellationToken`?

> As you saw in the tests, sometimes you are in a synchronous callback/override and need to do work.

The reason why I turned async-into-sync in the tests is to be able to re-use all synchronous tests for the async overloads (and avoid test duplication).

Reply:
> The reason I omitted sync overloads is because I think most of the stream scenarios can be satisfied with SignAsync/VerifyAsync (I may be wrong), 

Almost anything that someone thinks can be async-only ends up finding a need to be called from a synchronous context.  The written guideline is `CONSIDER adding synchronous variants of Async methods.` but it essentially means is "unless a synchronous version is straight up bonkers or can't sensibly be written, write one".